cmake failure: rocprofiler-sdk implicitly depends on rocdecode

After
```
cmake -B build -GNinja . -DTHEROCK_AMDGPU_FAMILIES=gfx1151
ninja -C build
```
I see
```
[rocprofiler-sdk configure] CMake Error at /scratch/mgehre/TheRock/build/media-libs/rocdecode/build/rocdecode-config.cmake:16 (include):
[rocprofiler-sdk configure]   include could not find requested file:
[rocprofiler-sdk configure] 
[rocprofiler-sdk configure]     /scratch/mgehre/TheRock/build/media-libs/rocdecode/build/rocdecode-targets.cmake
[rocprofiler-sdk configure] Call Stack (most recent call first):
[rocprofiler-sdk configure]   /scratch/mgehre/TheRock/cmake/therock_subproject_dep_provider.cmake:73 (find_package)
[rocprofiler-sdk configure]   cmake/Modules/Findrocdecode.cmake:30 (find_package)
[rocprofiler-sdk configure]   /scratch/mgehre/TheRock/cmake/therock_subproject_dep_provider.cmake:73 (find_package)
[rocprofiler-sdk configure]   cmake/rocprofiler_config_interfaces.cmake:368 (find_package)
[rocprofiler-sdk configure]   CMakeLists.txt:109 (include)
```

This might be because rocprofiler-sdk tries to find rocdecode (via rocm-systems/projects/rocprofiler-sdk/cmake/Modules/Findrocdecode.cmake), but the dependency on rocdecode is not represented in https://github.com/ROCm/TheRock/blob/176bfea06c646b573a6b902d8cd1b333d4b88008/profiler/CMakeLists.txt#L92.

It like like after this trace is triggered once, rocprofiler-sdk's CMakeCache.txt will set
`rocdecode_DIR=/scratch/mgehre/TheRock/build/media-libs/rocdecode/build`
instead of the correct `/scratch/mgehre/TheRock/build/media-libs/rocdecode/dist/lib/cmake/rocdecode`
which will make all subsequent builds also fail.

Please fix this instance of the problem and potentially provide some detection to avoid undeclared dependencies becoming races.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -496,6 +496,10 @@ add_subdirectory(compiler)
 add_subdirectory(third-party)
 add_subdirectory(core)
 add_subdirectory(debug-tools)
+# media-libs (rocdecode/rocjpeg) must be added before profiler: rocprofiler-sdk
+# performs optional find_package(rocdecode)/find_package(rocjpeg) for trace
+# feature detection.
+add_subdirectory(media-libs)
 # Note that rocprofiler-register is in base and is what higher level clients
 # depend on. The profiler itself is independent.
 add_subdirectory(profiler)
@@ -507,7 +511,6 @@ add_subdirectory(comm-libs)
 add_subdirectory(storage-libs)
 add_subdirectory(math-libs)
 add_subdirectory(ml-libs)
-add_subdirectory(media-libs)
 
 if(THEROCK_BUILD_TESTING)
   add_subdirectory(examples)

--- a/profiler/CMakeLists.txt
+++ b/profiler/CMakeLists.txt
@@ -6,6 +6,18 @@ if(THEROCK_ENABLE_MPI)
   set(_openmpi_optional_dep therock-openmpi)
 endif()
 
+# rocprofiler-sdk performs optional find_package(rocdecode)/find_package(rocjpeg)
+# to detect the rocDecode/rocJPEG trace features. Declare them as dependencies
+# when available so they resolve from the super-project dist rather than
+# escaping to a stray build tree.
+set(_rocprofiler_sdk_optional_deps)
+if(THEROCK_ENABLE_ROCDECODE AND THEROCK_ENABLE_SYSDEPS_AMD_MESA)
+  list(APPEND _rocprofiler_sdk_optional_deps rocdecode)
+endif()
+if(THEROCK_ENABLE_ROCJPEG AND THEROCK_ENABLE_SYSDEPS_AMD_MESA)
+  list(APPEND _rocprofiler_sdk_optional_deps rocjpeg)
+endif()
+
 if(THEROCK_ENABLE_ROCPROFV3)
 
   ##############################################################################
@@ -120,6 +132,7 @@ if(THEROCK_ENABLE_ROCPROFV3)
       rocprof-trace-decoder
       rocprofiler-register
       therock-yaml-cpp
+      ${_rocprofiler_sdk_optional_deps}
       ${THEROCK_BUNDLED_ELFUTILS}
       ${THEROCK_BUNDLED_LIBDRM}
       ${THEROCK_BUNDLED_SQLITE3}

